velox/exec/HashProbe.cpp

```diff
@@ -931,49 +934,50 @@ RowVectorPtr HashProbe::getOutputInternal(bool toSpillOutput) {
   }
   checkRunning();
 
-  if (!toSpillOutput) {
-    // Avoid memory reservation if it is triggered by memory arbitration to
-    // spill pending output.
-    ensureOutputFits();
-  }
-
   if (maybeReadSpillOutput()) {
     return output_;
   }
 
   clearIdentityProjectedOutput();
 
+  if (!toSpillOutput) {
```

Comment:
I assume no logic changing in this section? just refactoring, right?

Reply:
It avoids reserving memory if read input from spilled out as nothing can spill from this operator.

Reply:
Yeah also it made the structure cleaner. Thanks